spec/core.md

```diff
@@ -256,6 +256,241 @@ Finally, concatenate the prefix "did:peer:3" with the computed and encoded hash:
+    "#6MkrCD1c"
+  ],
+  "assertionMethod": [
+    "#6MkrCD1c"
+  ],
+  "keyAgreement": [
+    "#6LSqPZfn"
+  ],
+  "capabilityInvocation": [
+    "#6MkrCD1c"
+  ],
+  "capabilityDelegation": [
+    "#6MkrCD1c"
+  ],
+  "service": [
+    {
+      "id": "#didcommmessaging-0",
+      "type": "DIDCommMessaging",
+      "serviceEndpoint": "didcomm://queue",
+      "accept": [
+        "didcomm/v2"
+      ],
+      "routingKeys": []
+    }
+  ]
+}
+```
+To encode this value into a `did:peer:4`:
+
+1. Encode the document:
+    1. JSON stringify the object without whitespace
+    2. Encode the string as utf-8 bytes
+    3. Prefix the bytes with the multicodec prefix for json ([varint](https://github.com/multiformats/unsigned-varint) `0x0200`)
+    4. Multibase encode the bytes as base58btc (base58 encode the value and prefix with a `z`)
+    5. Consider this value the `encoded document`
+2. Hash the document:
+    1. Take SHA2-256 digest of the encoded document (encode the bytes as utf-8)
+    2. Prefix these bytes with the [multihash](https://github.com/multiformats/multihash) prefix for SHA2-256 and the hash length (varint `0x12` for prefix, varint `0x20` for 32 bytes in length)
+    3. Multibase encode the bytes as base58btc (base58 encode the value and prefix with a `z`)
+    4. Consider this value the `hash`
+3. Construct the did by concatenating the values as follows:
+
+        did:peer:4{{hash}}:{{encoded document}}
+
+Here is an example long form DID made from the input example above:
+
+```
+did:peer:4zQmNsz8npvrAyj983LTownQhp3PmGVGzMYrhBRGfig6rZ6P:z3U8mHxj7WY8i5EQD5rmkfPaDTws9dw5NRERbsFhsDFh5kvAEAY1afMFTWWbZT2d7TjCYtGBt47e1WpYoXYKSoEjxQX9qhNSw2ujMPiHn5rVyVCwQH1kBLzs95n37MoD6ikVDd29fkJoZaBrGhNaxcFiW9b1N85NWGQ9vs1LEMxsqt5ZHiFXGp1tMU85Da4VETb7veFZjvxRsadgbgeK1tFG83c6SSyDjoVzQbagXJ44kZXaMMmbs4Kk6VJszrCM1CUHPk7kFgHmojWU8kFAmPE88oesxBW5Wdc5cCNB6cB4QzRGuYrkK15hMdoFWq4kVHJn1xbruCzJ7Mb9f2rKUF6KTiUbTYpHuo4Kbnu26tJKQ9D7xHkAM2N3ZPN3eRkXBefKme5mLhGjRgXm6fAZiCHuK4dMyg4Bd9HDXiy8vSdY8cyZnyuJdPsjq5FRvRD92cFNtJZBJJwRQu6WiwKhTL9jELwwGfU2jukeESmARHjpQRTkXhtyG5NHDwj3Yx9CsbyBR5xdGsB3raA8JiMP4nAsbZhfXiBErBUx4MwYRnBDZERZztPjJWJniyKVG6hfoBokzEtkZt6gYMh1tpjsBAcSVw4C9H7o7QrY3mW6DjSufDdHSdWPVJjfHgRzxUM218CSiEwqEctqxJP9fc2FVSDxai7JUnroVzgYzhb62S4ueLGKM83abkd3Fm5NeSuewPRbgwETTLvknz1Wq1G4qygq75Fp3Kr21qknM2tsgrkwyprYR9ZTK5YzY5sHCwNP14VXZeX24QdSfevspNdvFtFtiDq6dUufmy5bKeLdHxx7Mpb7vFToU8bk9zZNUkcgXvX12U6iT1zLEyszTS6B3csHRr1HmvLUgEQKfd2aWjV2ScktEBsjZRHWdWuxQRcs85sF92kW2fVX7k1EGAwYGsnr6Wf9Q7jkM7SgJM5WJ1rHHsxKXwj8j11QmwXRcgREVEJphWdj87tpsC36A4rfEYhyJDw13UB68JgoK544NbsA
+```
+
+To construct the short form, simply omit the `:{{encoded document}}` from the end.
+
+Here is an example short form DID for the long form above:
+
+```
+did:peer:4zQmNsz8npvrAyj983LTownQhp3PmGVGzMYrhBRGfig6rZ6P
+```
+
+##### Resolving a DID
+
+###### Long form
+
+Resolving a long form `did:peer:4` document is done by decoding the document from the DID and "contextualizing" the document with the DID.
```

Comment:
I’d suggest reusing the defined term “Input Document” here to make it obvious that decoding gets you back to the original document.